845 recycle bin functionality

dstopp opened the pull request:
Recycle bin functionality -
Addresses issues:
 #845 
 #871  
 #864 
 #883 - not yet complete
Includes minor back end changes:
 -modified reset to return total counts/recycle counts
 -added option to return all the download option columns including items in recycle bin
Front end changes to #cart tab:
 -add recycle icon on obs image when observation in recycle bin
 -restore original edit ability on #cart tab, including ctrl-click and range options
 -change #cart add/remove icons and text to reflect remove/restore to recycle bin w/new icons in all places those icons appear, including the observation sliding menu, mini menu, gallery view and table view hover text.
 -in the table view, draw a green line through the observation to indicate in recycle bin
 -update the cart status badge to display on the number of items in the cart
 -move the download buttons to the download options panel
 -change the Empty Cart button to a nav option
 -Add to the nav line Empty Cart Recycle bin and Restore Recycle Bin to Cart w/badge showing number of items in the recycle bin
 -disable the Empty/Restore recycle bin nav items when there is nothing in the recycle bin
 -update the download options panel to reflect changes as observations are removed/restored to the cart, as well as the total size an total files counts
 -update Getting Started Guide

Additional changes:
 -added a border/shadow around the recycle and binoculars to make them stand out better;
 -made the cart border more defined by making rounded edges and thicker border

rfrenchseti committed:
Snapshot of implementation of cart recycle bin

rfrenchseti committed:
Recycle bin snapshot

rfrenchseti committed:
Finish implementation of backend recycle bin

dstopp committed:
Merge branch 'cart_recycle_bin_django' of https://github.com/rfrenchseti/pds-opus into 845_recycle_bin_functionality

# Conflicts:
#	opus/application/static_media/js/browse.js

dstopp committed:
Initial submit of code:
 - added back in edit functionality to cart page;
 - added recycle icon on cart page on;
 - updated the python code to return all product types including those in the recycled bin
 - updated reset.json to return recycled count
 - updated associated tests; one still failing
 - added green color and strikethru to table view to indicate recycled observation
 - moved download action nav items to download options panel
 - added new actions to the cart page to manage the recycle bin

dstopp committed:
-ongoing changes for the recycle bin
 - added shadow to the recycle icon
 - minor fixes to where/when the recycle icons show
 - enable edit on galleryView modal in #cart

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 845_recycle_bin_functionality

dstopp committed:
updated binoculars w/shadow

dstopp committed:
-updated icons and icon text on browse and cart page
-updated galleryView icons on #cart tab
-modified confirm wording

dstopp committed:
-add title for hover to recycle icon

dstopp committed:
-minor adjustment to the format on the download panel
-update the getting started text
-made the cart border more distinct
-update the text from toggle cart to more specific text
-update the gallery view modal when the recycle bin is emptied

rfrenchseti commented on Downloads/csv_pdsrms-data-h-2019-10-27T20-40-15.txt:
```diff
@@ -0,0 +1,1721 @@
+OPUS ID,Instrument Name,Planet,Intended Target Name,Observation Start Time,Observation Duration (secs)
```
I think this file was added by mistake. Please remove it and the Downloads directory.

rfrenchseti commented on opus/application/apps/cart/templates/cart/cart.html:
```diff
@@ -13,6 +13,22 @@ <h1>Download Options</h1>
             </div>
+            <div class="op-download-options">
+                <li class="list-inline-item">
+                    <a href="#" class="op-download-csv btn btn-outline-primary" download title="Download CSV of selected metadata for all observations in cart"><i class="fas fa-file-csv"></i>&nbsp;Metadata CSV</a>
```
For consistency with others, should say "Download *a* CSV". Maybe all should say "in the cart" instead of "in cart".  Likewise "using *the* selected products*.

rfrenchseti replied:
Still doesn't say "in the cart"

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -384,6 +373,28 @@ <h3 class='op-header-text'></h3>
+        {% include "ui/confirm_modal.html" with
+            id_name="op-empty-recycle-bin"
+            confirm_title="Confirm Empty Cart Recycle Bin"
+            confirm_msg="Are you sure you want to remove all observations from your recycle bin?"
```
Should say "the recycle bin"

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -384,6 +373,28 @@ <h3 class='op-header-text'></h3>
+        {% include "ui/confirm_modal.html" with
+            id_name="op-restore-recycle-bin"
+            confirm_title="Confirm Restore All Recycled Observations To Cart"
+            confirm_msg="Are you sure you want to restore all observations to your cart from the recycle bin?"
```
Sounds better as "from the recycle bin to your cart"

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -180,38 +180,27 @@ <h2>Your search produced no results</h2>
-                                        <a href="#" class="downloadData text-info" title="Download a zipped data archive of all observations in cart using selected products"><i class="fas fa-download"></i>&nbsp;Download Data Archive</a>
-                                    </div>
+                                    <a href="#" class="text-info mr-3" data-toggle="modal" data-target="#op-empty-recycle-bin" title="Remove all observations from recycle bin to cart"><i class="fas fa-recycle"></i>
+										&nbsp;Empty Cart Recycle Bin</a>
```
I think this should just say "Empty Recycle Bin"

rfrenchseti commented on opus/application/apps/help/templates/help/gettingstarted.html:
```diff
@@ -179,6 +179,17 @@ <h3>The Cart</h3>
     of observations, from the <b>Browse Results</b> tab. The number of
     observations in the cart will be displayed next to the <b>Cart</b> tab.</p>
 
+<p>Editing directly on the <b>Cart</b> tab will place the edited observations
```
(Look at this comment in preview mode to get the HTML markup in plain text)

How about this, since I don't know what "editing" means.

"While on the <b>Cart</b> tab, you can move observations to the recycle bin rather than removing them from the cart entirely. These observations will be marked with <recycle bin icon>. You can also restore observations from the recycle bin to the cart. Observations may be move to the recycle bin or restored to the cart as individual observations or as ranges of observations. In addition, all observations in the recycle bin can be moved to the cart at once by clicking on <b>Restore Recycle Bin to Cart</b>. All observations can be removed from the recycle bin by clicking on <b>Empty Recycle Bin</b>.

Observations in the recycle bin are not counted as part of the total result count, nor will they be included in the downloaded data. The total number of observations currently in the recycle bin will be displayed next to the <b>Restore Recycle Bin to Cart" option.

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -384,6 +373,28 @@ <h3 class='op-header-text'></h3>
         %}
+        {% include "ui/confirm_modal.html" with
+            id_name="op-empty-recycle-bin"
+            confirm_title="Confirm Empty Cart Recycle Bin"
```
Just "Confirm Empty Recycle Bin"

rfrenchseti reviewed (changes requested):
- Download data/options pane doesn't work when the browser is too short.
- When browser is narrow and download options pane is a fly-out, the button is still labeled "Download Options". It needs to say  something like "Download Data" to make it clear this is where you download the data.
- When the download options pane is a fly-out, it doesn't have the same format as it does when it's permanent. There are no headings for "Download Data" or "Download Options". This is because in the original version we moved "Download Options" to the header, but I don't think that's sufficient anymore.
- Likewise, when you narrow the browser, pop out the options pane, and then widen the browser so it becomes permanent, you don't get the same format as when it was permanent to start with.
- I'm having trouble reproducing it, but in one case of widening and narrowing the browser and popping out and hiding the download options pane, I got in a state where changing the contents of the cart no longer updated the product counts. In fact I've now gotten it to happen even without changing the browser size, just by moving between the browse results and cart tabs and adding and removing items. Aha I think I found a way to reproduce: Start with an empty cart, go to browse results, add an observation, go to cart, move the observation to the recycle bin, go to browse results, go to cart, move observation out of recycle bin. The product counts won't update.
- When using range on cart, popup should stay "Move range to recycle bin" or "Restore range from recycle bin"
- I find the three blue options in the nav bar really dim and hard to read. Also, since they're already dim, there's no indication when they are turned off because they don't dim further (unless you mouse over).
- I really like the recycle icon with the shadow, the binoculars with the shadow, and the thicker selection outline.
- I think the new thicker selection rectangle screwed up the size of the thumbnails. Reflow in the gallery no longer works smoothly as you widen and narrow the browser.

I did NOT review any of the JS changes.

dstopp committed:
-remove green bar from tools on table view when observation is in the recycle bin

dstopp committed:
-changed panel name "Download Options" to "Download Data" in all places including comments
-fixed the formating on the download data panel when collapsed/expanded
-updated the heading for the "Download Data" panel
-updated popup text for #cart tab
-really dimmed the blue nav options when not available
-removed Downloads/csv... file that was accidently added
-updated text in cart.html and base.html per comments
-changed "your cart/recycle bin" text style to "the cart/recycle bin" text style including confirm diaglog
-updated gettingstarted.html w/modified cart description of recycle bin

dstopp committed:
-updated test

dstopp committed:
make the 'dim' a bit less 'dim

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -180,42 +180,31 @@ <h2>Your search produced no results</h2>
                                 <ul class="sort-contents list-inline"></ul>
                             </div>
-                            <div class="op-download-options">
+							<div class="op-cart-actions">
```
This line appears to be indented improperly (or with tabs instead of spaces?)

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -180,42 +180,31 @@ <h2>Your search produced no results</h2>
+							<div class="op-cart-actions">
                                 <li class="list-inline-item">
-                                    <a href="#" class="op-download-csv text-info" download title="Download CSV of selected metadata for all observations in cart"><i class="fas fa-file-csv"></i>&nbsp;Download Metadata CSV</a>
+                                    <a href="#" class="text-info mr-3" data-toggle="modal" data-target="#op-empty-cart" title="Remove all observations from cart"><i class="fas fa-trash-alt"></i>
```
"from THE cart"

rfrenchseti commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -180,42 +180,31 @@ <h2>Your search produced no results</h2>
-                                    <div id="create_zip_data_file">
-                                        <a href="#" class="downloadData text-info" title="Download a zipped data archive of all observations in cart using selected products"><i class="fas fa-download"></i>&nbsp;Download Data Archive</a>
-                                    </div>
+                                    <a href="#" class="text-info mr-3" data-toggle="modal" data-target="#op-empty-recycle-bin" title="Remove all observations from recycle bin to cart"><i class="fas fa-recycle"></i>
```
Wrong tooltip? Remove all observations from the recycle bin.

dstopp committed:
minor updates to text

dstopp committed:
fix for the ugly wrap in the product display of download data panel

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -886,7 +909,7 @@ a:-webkit-any-link, a::any-link {
     pointer-events: none;
     cursor: default;
-    opacity: 0.1;
+    opacity: 0.25 !important;
```
How about 0.35? I tried it on a few different browsers and OSes and that seems to work better for me. It's still a big difference from the ~0.6 when it isn't disabled.

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 845_recycle_bin_functionality

# Conflicts:
#	log_analyzer/opus/slug.py
#	opus/application/settings.py

dstopp committed:
hide the download data panel when the screen is too short;
make the opacity of disabled things a bit less extreme;
update the text in confirm box

dstopp committed:
added the download data details to reset.json;
changed the break for short height;
fixed weird behaviour on browse tab when the recycle bin or cart is emptied

dstopp committed:
close the download data panel when the screen is too small.

dstopp committed:
update spacing on the slider

dstopp committed:
leave the slider on the first line when the cart is narrowed, doesn't need to go to second line.

dstopp committed:
fix text align on left when narrow after break at 700px

dstopp committed:
add commas to cart numbers;
change download text to "Download not Available" when the screen is too short

rfrenchseti committed:
Correctly computer number of items in the cart and recycle bin when deciding to throw a too big cart error. Fixes #629.

dstopp committed:
updated tests

rfrenchseti committed:
Merge branch '845_recycle_bin_functionality' of https://github.com/dstopp/pds-opus into recycle_bin_max_num

rfrenchseti committed:
Fix tests for cart counting

dstopp committed:
cleanup addComma code

dstopp committed:
Merge branch 'recycle_bin_max_num' of https://github.com/rfrenchseti/pds-opus into 845_recycle_bin_functionality

dstopp committed:
Add note about other product types to Download Options #883

dstopp committed:
update spinners on recover/empty recycle bin

dstopp committed:
set the cart spinner on empty/restore recycle bin

dstopp committed:
update tests for reset.json
